libs/compact-map/src/Data/Compact/KeyMap.hs

```diff
@@ -71,28 +106,32 @@ bitsPerSegment :: Int
 bitsPerSegment = 6
 {-# INLINE bitsPerSegment #-}
 
+bitmapInvariantMessage :: String -> Bitmap -> String
+bitmapInvariantMessage funcName b =
+  concat
+    [ "Bitmap ",
+      show b,
+      "has no bits set in '",
+      funcName,
+      "', this violates the bitmap invariant."
+    ]
+{-# NOINLINE bitmapInvariantMessage #-}
+
 -- | Ints in the range [0.. intSize], represents one 'bitsPerSegment' wide portion of a key
 type Segment = Int
 
 -- | Represents a list of 'Segment', which when combined is in 1-1 correspondance with a Key
-type Path = [Segment]
-
--- | The maximum value of a Segment, as an Int
-intSize :: Int
-intSize = 2 ^ bitsPerSegment
-{-# INLINE intSize #-}
+type Path = VP.Vector Segment
 
 -- | The maximum value of a segment, as a Word64
 wordSize :: Word64
```

Comment:
A little confusingly named! But I guess it's not exported

Reply:
Very good point. I've renamed it to `segmentMaxValue`